File: AdventurePlannerBE/ViewModels/ActivityDTO.cs
using AdventurePlannerBE.Enums;
using AdventurePlannerBE.Models;
using System.ComponentModel.DataAnnotations;

namespace AdventurePlannerBE.ViewModels
{
    public class ActivityDTO
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Location { get; set; }
        public DateOnly Date { get; set; }
        public Guid TripId { get; set; }
        public float Rating { get; set; }
        public int RatingCounts { get; set; }
        public string WebsiteUri { get; set; }
        public bool GoodForChildren { get; set; }
        public PriceLevel PriceLevel { get; set; }

        public ActivityDTO MapData(Activity activity)
        {
            Id = activity.Id;
            Name = activity.Name;
            Date = activity.Date;
            Location = activity.Location;
            TripId = activity.TripId;
            Rating = activity.Rating;
            RatingCounts = activity.RatingCounts;
            WebsiteUri = activity.WebsiteUri;
            GoodForChildren = activity.GoodForChildren;
            PriceLevel = activity.PriceLevel;

            return this;
        }
    }
}

File: AdventurePlannerFE/src/app/dtos/activity-dto.ts
import { EMPTY_GUID } from '../constants';
import { PriceLevel } from '../pages/models/enums/price-level';
import { SuggestedActivityViewModel } from '../pages/models/suggested-activity-view-model';

export class ActivityDto {
  id = '';
  name = '';
  location = '';
  date: string | null = null;
  rating = 0;
  ratingCounts = 0;
  websiteUri = '';
  goodForChildren = false;
  tripId = '';
  priceLevel: PriceLevel = 0;

  updateBasedOnModel(model: SuggestedActivityViewModel): ActivityDto {
    let result = new ActivityDto();
    if (model.id == '') {
      this.id = EMPTY_GUID;
    } else {
      this.id = model.id;
    }

    this.name = model.name;
    this.location = model.location;
    if (model.date) {
      this.date = model.date.toISOString().split('T')[0];
    }
    this.rating = model.rating;
    this.ratingCounts = model.ratingCounts;
    this.websiteUri = model.websiteUri;
    this.goodForChildren = model.goodForChildren;
    this.tripId = model.tripId;
    this.priceLevel = model.priceLevel;

    return result;
  }
}
